Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm

```diff
@@ -136,18 +140,43 @@ - (void)layoutSublayers
         if (auto* playbackSessionModel = this->playbackSessionModel()) {
             playbackSessionModel->setSpatialTrackingLabel(m_spatialTrackingLabel);
             playbackSessionModel->setSoundStageSize(WebCore::AudioSessionSoundStageSize::Large);
+            m_savedAutoDimmingPreference = playableViewController().prefersAutoDimming;
+
+            if (RefPtr model = this->videoPresentationModel())
+                playableViewController().prefersAutoDimming = model->prefersAutoDimming();
+
+            m_savedAutoDimmingPreference = playableViewController().prefersAutoDimming;
+
         }
         completionHandler(success, error);
     }).get()];
 }
 
+void VideoPresentationInterfaceLMK::prefersAutoDimmingChanged(bool prefersAutoDimming)
+{
+    if (prefersAutoDimming == m_savedAutoDimmingPreference)
+        return;
+
+    if (RetainPtr playerController = m_playerViewController) {
+        [playerViewController setPrefersAutoDimming:prefersAutoDimming];
+        m_savedAutoDimmingPreference = [playerViewController prefersAutoDimming];
```

Comment:
can the value changed between the time it is set and read again?

what about just:
```suggested
    if (std::exchange(m_savedAutoDimmingPreference, prefersAutoDimming) != prefersAutoDimming)
        [m_playerViewController setPrefersAutoDimming:prefersAutoDimming];
```

there's no need to test for nil with objC objects.